types/coin_test.go

```diff
@@ -391,10 +400,15 @@ func TestCoins(t *testing.T) {
 		{"mineral", NewInt(1)},
 		{"tree", NewInt(1)},
 	}
+	goodCaps := Coins{
+		{"GAS", NewInt(1)},
+		{"MINERAL", NewInt(1)},
+		{"TREE", NewInt(1)},
+	}
 	mixedCase1 := Coins{
-		{"gAs", NewInt(1)},
 		{"MineraL", NewInt(1)},
 		{"TREE", NewInt(1)},
+		{"gAs", NewInt(1)},
```

Comment:
I guess this test enforces the ordering by denom as published.

I just think if you normalize cases one place, they should be normalized everywhere.

Also, what if we have `bal := Coins{{"atom", NewInt(10)}}` and then `bal.Sub(Coins{{"ATOM", NewInt(3)}})`. What is the expected behavior?

Either `"atom" == "ATOM"` everywhere or nowhere - if they are considered dups, they should be treated the same by all math functions.

Reply:
I thought `"atom"  != "ATOM"` everywhere? Is this not correct @fedekunze ?

Reply:
yes, I removed it now